assets/js/googlesitekit/datastore/user/surveys.js

```diff
@@ -0,0 +1,154 @@
+	},
+	argsToParams: ( triggerID ) => {
+		return { triggerID };
+	},
+	reducerCallback: ( state, { survey_payload, session } ) => { // eslint-disable-line camelcase
+		return {
+			...state,
+			currentSurvey: survey_payload,
+			currentSurveySession: session,
+		};
+	},
+	validateParams: ( { triggerID } = {} ) => {
+		invariant( 'string' === typeof triggerID, 'triggerID is mandatory and must be a string' );
+	},
+
+} );
+
+const fetchSendSurveyEventStore = createFetchStore( {
+	baseName: 'sendSurveyEvent',
+	controlCallback: ( { event, session } ) => API.set( 'core', 'user', 'survey-event', { event, session } ),
+	argsToParams: ( event, session ) => {
+		return { event, session };
+	},
+} );
+
+const baseInitialState = {
+	currentSurvey: null,
+	currentSurveySession: null,
+};
+
+const baseActions = {
+	triggerSurvey: createValidatedAction(
+		( triggerID, options = { ttl: 0 } ) => {
+			invariant( 'string' === typeof triggerID, 'triggerID is mandatory and must be a string' );
+			invariant( isPlainObject( options ), 'options must be an object' );
+			invariant( 'number' === typeof options.ttl, 'options.ttl must be a number' );
+		},
+		function* ( triggerID, options = { ttl: 0 } ) {
+			const { select } = yield Data.commonActions.getRegistry();
+			if ( null !== select( STORE_NAME ).getCurrentSurvey() ) {
+				return;
+			}
+			const cacheKey = createCacheKey( 'core', 'user', 'survey-event', { triggerID } );
+			const { cacheHit, value } = yield Data.commonActions.await( getItem( cacheKey ) );
+			if ( false === cacheHit && options.ttl ) {
+				const { error, response } = yield fetchTriggerSurveyStore.actions.fetchTriggerSurvey( triggerID );
+				if ( ! error && options.ttl > 0 ) {
+					yield Data.commonActions.await( setItem( cacheKey, {} ) );
+					return { response, error };
+				}
+			}
+			return {
+				response: value,
+				error: false,
+			};
+		}
+	),
+	sendSurveyEvent: createValidatedAction(
+		( eventID, eventData = {} ) => {
+			invariant( 'string' === typeof eventID, 'eventID is mandatory and must be a string' );
+			invariant( isPlainObject( eventData ), 'eventData must be an object' );
+		},
+		function* ( eventID, eventData = {} ) {
+			const event = { [ eventID ]: eventData };
+			const { select } = yield Data.commonActions.getRegistry();
+			const session = select( STORE_NAME ).getCurrentSurveySession();
+			if ( session ) {
+				const { response, error } = yield fetchSendSurveyEventStore.actions.fetchSendSurveyEvent( event, session );
+				return { response, error };
+			}
+		}
+	),
+};
+
+const baseSelectors = {
+	/**
+	 * Gets the current survey.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {Object} Current survey object.
+	 */
+	getCurrentSurvey( state ) {
+		return state.currentSurvey;
+	},
+	/**
+	 * Gets the current survey session.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {Object} Current survey object.
```

Comment:
this comment is the same as the return comment above. I imagine it should be different? :thinking: